src/OpenSage.Tools.BigEditor/UI/MainForm.cs

```diff
@@ -253,53 +301,91 @@ private void DrawFilesList()
 
                 ImGui.NextColumn();
 
-                ImGui.Text(entry.Length.ToString());
+                ImGui.Text(ImGuiUtility.GetFormatedSize(entry.Length));
                 ImGui.NextColumn();
 
-                var exportId = "Export##ExportDialog" + i;
                 if (shouldOpenSaveDialog)
                 {
-                    ImGui.OpenPopup(exportId);
+                    var saveDialog = new SaveFileDialog("Export file");
+                    saveDialog.DefaultFileName = entry.Name;
+                    saveDialog.Save(result => ExportFile(entry, result.FileName));
                 }
+            }
 
-                bool contextMenuOpen = true;
-                if (ImGui.BeginPopupModal(exportId, ref contextMenuOpen, ImGuiWindowFlags.AlwaysAutoResize))
-                {
-                    ImGuiUtility.InputText("File Path", _filePathBuffer, out var filePath);
+            ImGui.Columns(1, null, false);
 
-                    if (ImGui.Button("Save"))
-                    {
-                        filePath = ImGuiUtility.TrimToNullByte(filePath);
+            ImGui.EndChild();
 
-                        using (var entryStream = entry.Open())
-                        {
-                            using (var fileStream = File.OpenWrite(filePath))
-                            {
-                                entryStream.CopyTo(fileStream);
-                            }
-                        }
+            ImGui.EndChild();
+        }
 
-                        ImGui.CloseCurrentPopup();
-                    }
+        private void DrawStatusPanel()
+        {
+            ImGui.BeginChild("statusbar", new Vector2(0, 30), true, 0);
+
+            ImGui.Text($"{_bigArchive.FilePath} | Version: {_bigArchive.Version} | Size: {ImGuiUtility.GetFormatedSize(_bigArchive.Size)} | Files: {_bigArchive.Entries.Count}");
+
+            ImGui.SameLine();
 
-                    ImGui.SetItemDefaultFocus();
+            if (_currentFileName.CompareTo("") != 0)
+            {
+                ImGui.Text($"| Selected file: {_currentFileName}");
+            }
 
-                    ImGui.SameLine();
+            ImGui.EndChild(); // end statusbar
+        }
 
-                    if (ImGui.Button("Cancel"))
+        private void Export(string path)
+        {
+            foreach (var entry in _files)
+            {
+                string filePath = $"{path}{Path.DirectorySeparatorChar}{Path.GetDirectoryName(entry.FullName.Replace('\\', Path.DirectorySeparatorChar))}";
```

Comment:
This is probably a good place to use [`Path.Combine`](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netstandard-2.1)

Reply:
Replaced